aspnet/2-structured-data/Models/DatastoreBookStore.cs

```diff
@@ -29,6 +35,8 @@ public static class DatastoreBookStoreExtensionMethods
         /// <returns>A datastore key.</returns>
         public static Key ToKey(this long id)
         {
+            // P0 This function is especially useful, but I see no way to implement it
+            // with the new KeyFactory().  Because this is static.
             return new Key()
             {
                 Path = new KeyPathElement[]

```

Comment:
I'm surprised this even compiles, to be honest - must look into why it does. I'd expect you to use

```csharp
new Key
{
    Path = { new KeyPathElement { ... } }
}
```